core/src/main/kotlin/org/evomaster/core/search/gene/FloatGene.kt

```diff
@@ -46,24 +61,36 @@ class FloatGene(name: String,
             }
         }
 
-        //TODO min/max for Float
-        val k = when (randomness.choose(listOf(0, 1, 2))) {
-            //for small changes
-            0 -> value + randomness.nextGaussian()
-            //for large jumps
-            1 -> value + (GeneUtils.getDelta(randomness, apc) * randomness.nextGaussian())
-            //to reduce precision, ie chop off digits after the "."
-            2 -> BigDecimal(value.toDouble()).setScale(randomness.nextInt(15), RoundingMode.HALF_EVEN).toDouble()
-            else -> throw IllegalStateException("Regression bug")
+        /*
+            TODO min/max for Float
+            Man: update a bit by considering min and max,
+            NEED a check by Andrea
+         */
+        var gaussianDelta = randomness.nextGaussian()
+        if ((max != null && max == value && gaussianDelta > 0) || (min != null && min == value && gaussianDelta < 0) )
+            gaussianDelta *= -1.0
+
+        val maxRange = if (!isRangeSpecified()) Long.MAX_VALUE
+        else if (gaussianDelta > 0)
+            NumberCalculationUtil.calculateIncrement(value.toDouble(), (max?: Float.MAX_VALUE).toDouble()).toLong()
+        else
+            NumberCalculationUtil.calculateIncrement((min?: Float.MIN_VALUE).toDouble(), value.toDouble()).toLong()
+
+        var res = modifyValue(randomness, value.toDouble(), delta = gaussianDelta, maxRange = maxRange, specifiedJumpDelta = GeneUtils.getDelta(randomness, apc, maxRange),precision == null).toFloat()
+
+        if (precision != null && getFormattedValue() == getFormattedValue(res)){
+            res += (if (gaussianDelta>0) 1.0f else -1.0f) * getMinimalDelta()!!
         }
 
-        value = k.toFloat()
+        value = if (max != null && res > max) max
+                else if (min != null && res < min) min
+                else getFormattedValue(res)
 
         return true
     }
```

Comment:
hmmm... here code is similar to Double. is there any way in which code could be re-used? (ie avoiding duplication)